src/libcore/slice.rs

```diff
@@ -655,6 +655,12 @@ impl<'a, T> Default for &'a [T] {
     fn default() -> &'a [T] { &[] }
 }
 
+#[stable(feature = "mut_slice_default", since = "1.5.0")]
+impl<'a, T> Default for &'a mut [T] {
+    #[stable(feature = "mut_slice_default", since = "1.5.0")]
```

Comment:
@Gankro The argument about generics is equally applicable to annotations on impls (everything is even worse when trait objects are involved), but stability in concrete code is still can be respected in both cases, I guess. Not sure how useful is this though.